src/imex.rs

```diff
@@ -512,120 +622,89 @@ async fn import_backup(context: &Context, backup_to_import: impl AsRef<Path>) ->
 /*******************************************************************************
  * Export backup
  ******************************************************************************/
-/* the FILE_PROGRESS macro calls the callback with the permille of files processed.
-The macro avoids weird values of 0% or 100% while still working. */
+#[allow(unused)]
 async fn export_backup(context: &Context, dir: impl AsRef<Path>) -> Result<()> {
     // get a fine backup file name (the name includes the date so that multiple backup instances are possible)
-    // FIXME: we should write to a temporary file first and rename it on success. this would guarantee the backup is complete.
-    // let dest_path_filename = dc_get_next_backup_file(context, dir, res);
     let now = time();
-    let dest_path_filename = dc_get_next_backup_path(dir, now).await?;
-    let dest_path_string = dest_path_filename.to_string_lossy().to_string();
+    let (temp_path, dest_path) = get_next_backup_path_new(dir, now).await?;
 
+    context
+        .sql
+        .set_raw_config_int(context, "backup_time", now as i32)
+        .await?;
     sql::housekeeping(context).await;
 
-    context.sql.execute("VACUUM;", paramsv![]).await.ok();
+    context
+        .sql
+        .execute("VACUUM;", paramsv![])
+        .await
+        .map_err(|e| async move {
+            info!(context, "Vacuum failed, trying different working dir {}", e);
+            let pwd = env::current_dir();
+            env::set_current_dir(
```

Comment:
Since this fix, it does not vacuum, but does not output any errors too. Maybe some `.await` is missing? Anyway, let's just revert this attempt to change dir and vacuum again, if it fails it is not a critical problem, and it only fails on my phone.

Reply:
Did you check if a transaction is pending?
I saw the case that _vacuum_ with an exported backup file wasn't possible immediately, but it ran after launching sql command "end transaction" before trying _vacuum_.

Reply:
We can discuss this in https://github.com/deltachat/deltachat-core-rust/issues/1756.